### PR TITLE
Add Haxe 4.2.3 to CI

### PR DESCRIPTION
**Pull request by anissen**

### CODE REVIEW

File: .github/workflows/instrument.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
   build:
     strategy:
       matrix:
-        haxe-version: ['4.1.5', '4.2.1', 'nightly']
+        haxe-version: ['4.1.5', '4.2.1', '4.2.3', 'nightly']
```
Comment (AlexHaxe) [inline .github/workflows/instrument.yml]: I think we can drop 4.2.1 entirely

Comment (anissen) [reply .github/workflows/instrument.yml]: Done. You can choose to squash the commits if you wish.

**Comment by AlexHaxe** [comment]
thanks!

**Comment by anissen** [comment]
According to [CodeCov.io](https://codecov.io/gh/AlexHaxe/haxe-instrument/commit/6aa9a45ab7c093ec75ac42dc2887c7878fc56da9/) this PR reduces code coverage by 3.63% 🤔

**Comment by AlexHaxe** [comment]
it went down to the same level before last commit.
I'm not sure why last run increased coverage that much or why it wouldn't be the same after merging your PR.

**Pull request merged**